src/battle_ai_util.c

```diff
@@ -605,11 +605,25 @@ struct SimulatedDamage AI_CalcDamage(u32 move, u32 battlerAtk, u32 battlerDef, u
         }
         else
         {
-            s32 nonCritDmg = CalculateMoveDamageVars(move, battlerAtk, battlerDef, moveType, fixedBasePower,
-                                                    effectivenessMultiplier, weather, FALSE,
-                                                    aiData->holdEffects[battlerAtk], aiData->holdEffects[battlerDef],
-                                                    aiData->abilities[battlerAtk], aiData->abilities[battlerDef]);
+            s32 nonCritDmg = 0;
+            if (gMovesInfo[move].effect == EFFECT_TRIPLE_KICK)
+            {
+                for (gMultiHitCounter = 3; gMultiHitCounter > 0; gMultiHitCounter--)
```

Comment:
I may be missing something trivial, but is there a reason we're using a global here instead of a local variable?

Reply:
yes, the global is needed for this part
```
case EFFECT_TRIPLE_KICK:
    basePower *= 1 + gMovesInfo[move].strikeCount - gMultiHitCounter;
    break;
```
Usually what happens is the at the start of the move execution `gMultiHitCounter` is set for multi hit moves and decremented for each hit. There was previously a had that just multiplied the triple kick damage by 6 to approximate it but it was not accurate.

What this change does is simulate it with `gMultiHitCounter`

Reply:
Shouldn't `gMultiHitCounter = 3` be `gMultiHitCounter = gMovesInfo[move].strikeCount` then?

Reply:
I can add a comment just in case somebody else is surprised